### PR TITLE
Added support for ConfigMap, and Volumes

### PR DESCRIPTION
**Pull request by rahulchheda**

### CODE REVIEW

**Review by Sanjay1611 — CHANGES_REQUESTED** [review]
Please add the comments to functions in `/configmap/build.go`

**Comment by Sanjay1611** [comment]
You have added the `configmap` builder in #6  also

**Comment by rahulchheda** [comment]
> You have added the `configmap` builder in #6 also

Will be over-ridden once that PR is merged!

File: kubernetes/configmap/build.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package configmap
+
+import (
+	"errors"
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
```
Comment (chandankumar4) [inline kubernetes/configmap/build.go:23]: Can you make the seperation between, different package type

File: kubernetes/podtemplatespec/podtemplatespec.go [hunk 2]
```diff
@@ -340,6 +340,35 @@ func (b *Builder) WithContainerBuilders(
 	return b
 }
 
+// WithVolumeBuilders builds the list of volumebuilders provided
+// and merges it to the volumes field of podtemplatespec.
+func (b *Builder) WithVolumeBuilders(volumeBuilderList []*volume.Builder) *Builder {
```
Comment (chandankumar4) [inline kubernetes/podtemplatespec/podtemplatespec.go:346]: This function is already in this file with commented, can you remove that

File: kubernetes/volume/v1alpha1/build.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	errors "github.com/openebs/maya/pkg/errors/v1alpha1"
```
Comment (chandankumar4) [inline kubernetes/volume/v1alpha1/build.go]: why are we using the package from maya?

File: kubernetes/volume/v1alpha1/build.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2019 The OpenEBS Authors
```
Comment (ksatchit) [inline kubernetes/volume/v1alpha1/build.go]: Need to change this? or retain? @chandankumar4

**Review by chandankumar4 — CHANGES_REQUESTED** [review]
Have some comments ptal

File: kubernetes/volume/v1alpha1/build.go (hunk 4 above)
Comment (chandankumar4) [reply kubernetes/volume/v1alpha1/build.go]: need to change this

File: kubernetes/volume/v1alpha1/build.go (hunk 3 above)
Comment (ksatchit) [reply kubernetes/volume/v1alpha1/build.go]: Was thinking of this too -- either use all from maya OR use all from here, I suppose, for the builder context

Comment (chandankumar4) [reply kubernetes/volume/v1alpha1/build.go]: if we use all from maya then some of the part is not there as @amitbhatt818 mentioned earlier then we have to send PR to maya then, So I think better to use from here only

**Review by chandankumar4 — APPROVED** [review]

**Review by ksatchit — APPROVED** [review]

**Pull request merged**